### PR TITLE
draw laser line

### PR DESCRIPTION
**Pull request by gustavoneves12**
initial code to draw laser line

### CODE REVIEW

File: viz/LaserLine.cpp [hunk 1]
```diff
@@ -3,40 +3,102 @@
...
+
+    /**
+     * add the primitive set to geometry
+     */
+    geometry->addPrimitiveSet(primitiveSet);
+
+    /**
+     * set line colors
+     */
+    colors->push_back(osg::Vec4(0.1, 0.9, 0.1, 1.0));
```
Comment (doudou) [inline viz/LaserLine.cpp]: Make the color configurable. Please.

File: viz/LaserLine.cpp [hunk 2]
```diff
@@ -3,40 +3,102 @@
...
+
+    //laser scan depth values
+    std::vector<uint32_t> ranges = data.ranges;
+
+    double angle = data.start_angle; //start angle
+    double angleStep = data.angular_resolution; //angle step
+    double maxRange = data.maxRange / 1000.0; //convert millimeters to meters
+
+    vertices->clear();
+    for (int i = 0; i < ranges.size()-1; i++) {
```
Comment (doudou) [inline viz/LaserLine.cpp]: This will lead to very bad things if the ranges are empty.

**Comment by gustavoneves12** [comment]
PING

File: viz/LaserLine.cpp [hunk 3]
```diff
@@ -3,40 +3,137 @@
...
+    //laser scan depth values
+    std::vector<uint32_t> ranges = data.ranges;
+
+    double angle = data.start_angle; //start angle
+    double angleStep = data.angular_resolution; //angle step
+    double maxRange = data.maxRange / 1000.0; //convert millimeters to meters
+
+    vertices->clear();
+
+    if (showRange) {
```
Comment (doudou) [inline viz/LaserLine.cpp:67]: Remove this, please. There is already a plugin to display the ranges for a laser scanner, really no need to duplicate the functionality (at the cost of making this plugin more complicated).

Comment (doudou) [inline viz/LaserLine.cpp:67]: I'm merging it anyways, but please keep it in mind.

**Pull request merged**